src/Pages/Auth/Register.jsx

```diff
@@ -1,19 +1,93 @@
-import React, { useState, useEffect, useRef } from "react";
-import { Link } from "react-router-dom";
+import React, { useEffect, useState } from "react";
+import { Link, Redirect } from "react-router-dom";
 import logo from "../../assets/budget-it.png";
 import { currency } from "./currency";
 import { TimelineLite } from "gsap";
+import { useFormik } from "formik";
+import * as Yup from "yup";
+import axios from "axios";
 
-const Register = () => {
-  const [user, setUser] = useState({
-    firstname: "",
-    lastname: "",
-    email: "",
-    password: "",
-    confirmPassword: "",
-    currency: "NGN",
+const Register = (props) => {
+  const [validationError, setValidationError] = useState(null);
+  const [user, setUser] = useState(null);
+  if (user) {
+    props.history.push("/dashboard");
+    localStorage.setItem("user", JSON.stringify(user.token));
+  }
+  const formik = useFormik({
+    initialValues: {
+      firstname: "",
+      lastname: "",
+      email: "",
+      password: "",
+      confirmPassword: "",
+      currency: "NGN",
+    },
+
+    validationSchema: Yup.object({
+      firstname: Yup.string()
+        .min(2, "First name must be greater than 2 characters")
+        .required("Required")
+        .matches(
+          /[A-Za-z0-9 ]+$/,
+          "First name must  not contain special characters"
+        ),
+
+      lastname: Yup.string()
+        .min(2, "Last name must be greater than 2 characters")
+        .required("Required")
+        .matches(
+          /[A-Za-z0-9 ]+$/,
+          "Last name must not a contain special character"
+        ),
+
+      email: Yup.string()
+        .matches(
+          /^[a-z0-9][a-z0-9-_.]+@([a-z]|[a-z0-9]?[a-z0-9-]+[a-z0-9])\.[a-z0-9]{2,10}(?:\.[a-z]{2,10})?$/,
+          "Email must be valid"
+        )
+        .required("Required"),
+
+      password: Yup.string()
+        .min(6, "Password is too short")
+        .required("Required"),
+
+      confirmPassword: Yup.string()
+        .min(6, "Password is too short")
+        .oneOf([Yup.ref("password")], "Passwords do not match"),
+
+      currency: Yup.string().required("Required or default set to NGN"),
+    }),
+
+    onSubmit: (values) => {
+      axios({
+        method: "post",
+        url: "https://atumaatu.herokuapp.com/v1/users",
+        data: JSON.stringify({
+          user: {
+            firstname: values.firstname,
+            lastname: values.lastname,
+            email: values.email,
+            password: values.password,
+            password_confirmation: values.confirmPassword,
+            currency: values.currency,
+          },
+        }),
+        headers: {
+          "Content-Type": "application/json",
+        },
+      })
+        .then(function (response) {
+          setUser(response.data.payload);
+        })
+        .catch(function (error) {
+          if (error && error.response) {
+            setValidationError(error.response.data.error);
```

Comment:
This should be authentication error not validationError. You have yup and formik taking care of validation errors

Reply:
Ohk